panel/chat/feed.py

```diff
@@ -428,16 +430,31 @@ async def _prepare_response(self, _) -> None:
 
         disabled = self.disabled
         try:
-            self.disabled = True
+            with param.parameterized.batch_call_watchers(self):
+                self.disabled = True
+                self._callback_is_running = True
+
             message = self._chat_log[-1]
             if not isinstance(message, ChatMessage):
                 return
 
             num_entries = len(self._chat_log)
-            task = asyncio.create_task(self._handle_callback(message))
-            await self._schedule_placeholder(task, num_entries)
-            await task
-            task.result()
+            loop = asyncio.get_event_loop()
+            contents = self._extract_contents(message)
+
+            if asyncio.iscoroutinefunction(self.callback):
+                future = loop.create_task(self._wrap_async_callback(contents, message))
+            else:
+                future = loop.run_in_executor(None, partial(self.callback, contents, message.user, self))
```

Comment:
I think in future I want this to go via `state.execute` but that needs to be rewritten to return a `Future` and to allow forcing threaded execution even when there isn't a ThreadPoolExecutor available.